pallets/uniques/src/lib.rs

```diff
@@ -0,0 +1,755 @@
+/*
+ * This file is part of the Nodle Chain distributed at https://github.com/NodleCode/chain
+ * Copyright (C) 2020-2022  Nodle International
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+//! Handle the ability to notify other pallets that they should stop all
+
+use frame_support::traits::{Currency, ExistenceRequirement, ReservableCurrency};
+pub use pallet::*;
+use pallet_uniques::{DestroyWitness, WeightInfo as UniquesWeightInfo};
```

Comment:
This import is difficult to read. It is used as 
```
      #[pallet::weight(<T as pallet_uniques::Config<I>>::WeightInfo::buy_item())]

```

Reply:
You need to bring the trait `WeightInfo` from the pallet_uniques into the scope without a name collision with this pallet's own WeightInfo. That's the solution we have.